synapse/app/homeserver.py

```diff
@@ -315,9 +315,11 @@ def start_listening(self, listeners):
                         )
                     )
                 else:
-                    _base.listen_metrics(listener["bind_addresses"], listener["port"])
+                    _base.listen_metrics(listener.bind_addresses, listener.port)
             else:
-                logger.warning("Unrecognized listener type: %s", listener["type"])
+                # this shouldn't happen, as the listener type should have been checked
+                # during parsing
+                logger.warning("Unrecognized listener type: %s", listener.type)
```

Comment:
Did you want to keep this consistent with the modified message for workers ("Unsupported...")?

Reply:
well... here we know that if we get to the end of the list, it's an unknown listener type. In the workers, it could be a *known* type which is unsupported on workers.

I don't know if that's a good reason, but it was the reasoning...

Reply:
Ah, yes that makes sense! 👍 I think keeping them different is fine.